example/flutter_example/lib/search_page.dart

```diff
@@ -0,0 +1,246 @@
+import 'package:faker/faker.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_example/book.dart';
+import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
+import 'package:meilisearch/meilisearch.dart';
+import 'dart:async';
+
+import 'highlighted_text.dart';
+
+class SearchPage extends StatefulWidget {
+  const SearchPage({super.key});
+
+  @override
+  State<SearchPage> createState() => _SearchPageState();
+}
+
+class _SearchPageState extends State<SearchPage> {
+  static const _pageSize = 20;
+  //If you are using this on an android emulator, check this: https://developer.android.com/studio/run/emulator-networking
+  final client = MeiliSearchClient('http://127.0.0.1:7700', 'masterKey');
+  MeiliSearchIndex? index;
+  final _textController = TextEditingController();
+  final faker = Faker();
+  bool isPagination = false;
+  int? totalHits;
+  final _pagingController = PagingController<int, PersonDto>(firstPageKey: 0);
+  Future<MeiliSearchIndex>? _initFuture;
+  static const kPreTag = '<b>';
+  static const kPostTag = '</b>';
+  Future<void> seedFakeBooks({int count = 100}) async {
+    if (index == null) return;
+    final task = await index!.addDocuments(
+      Iterable.generate(
+        100,
+        (index) {
+          final id = faker.guid.guid();
+          final name = faker.person.name();
+          //we duplicate the name to simulate cases where there are multiple highlights
+          return PersonDto(id: id, name: "$name $name");
+        },
+      ).map((e) => e.toMap()).toList(),
+      primaryKey: PersonDto.kid,
+    );
+    //Usually you would need to await for the task to complete, see https://github.com/meilisearch/meilisearch-dart/issues/260
+    await Future.delayed(const Duration(milliseconds: 100));
+    _pagingController.refresh();
+  }

```

Comment:
I think you could move this somewhere else :D

Reply:
do I make a separate file, "index.dart" ?

Reply:
Or index_setup.dart